cmd/samples/pso/workflow_test.go

```diff
@@ -39,14 +39,16 @@ func Test_Workflow(t *testing.T) {
 			panic("unexpected activity call")
 		}
 	})
+
+	var childWorkflowID string
 	env.SetOnChildWorkflowStartedListener(func(workflowInfo *workflow.Info, ctx workflow.Context, args encoded.Values) {
-		queryAndVerify(t, env, "child", "PSO_child_workflow:default-test-run-id") // Child WorkflowID
+		childWorkflowID = workflowInfo.WorkflowExecution.ID
 	})
 
 	env.ExecuteWorkflow(PSOWorkflow, "sphere")
 
 	require.True(t, env.IsWorkflowCompleted())
-	//queryAndVerify(t, env, "child", "PSO_child_workflow:default-test-run-id_RunID") // Child WorkflowID+RunID
+	queryAndVerify(t, env, "child", childWorkflowID)
 	require.Equal(t, env.GetWorkflowError().Error(), "ContinueAsNew") // consider recreating a new test env on every iteration and calling execute workflow with the arguments from the previous iteration (contained in ContinueAsNewError)
```

Comment:
Is it possible to test a child workflow?
And is it possible to query a child workflow?
It seems to me that there is no support for it, am i missing something?

Reply:
I believe child workflow testing is supported. Have you tried querying it?

Reply:
Let me rephrase it, the child workflow returns a ContinueAsNewError that I can get with env.GetWorkflowError() but I cannot cast it to the structure to extract the data necessary to run another child iteration.

Reply:
also env.QueryWorkflow() is tied to the parent workflow as it does not recognise the query in the child. Should I create a new env for the child and call env.ExecuteWorkflow(PSOChildWorkflow, ...)?

Reply:
I see. I think we miss some features in child workflow testing.